streamlit_app_tutorial/streamlit_demo_iod_analysis.py

```diff
@@ -12,23 +12,29 @@
 from getters.lsoa_shapefiles_2011 import get_english_lsoa_shapefiles_2011
 import os
 
+# Make sure to read the README.md file for this app before running this code.
+
 
 # Caches the data to prevent computation on every rerun
 @st.cache_data
```

Comment:
You can do @st.cache_data(show_spinner = "Loading data") to specify the message that the user sees while the app is spinning up

I can't comment directly on this line, but on line 165, I would put the steps like this that manipulate the data inside a function that's cached